src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/adjudications/AdjudicationsMigrationService.kt

```diff
@@ -310,20 +311,37 @@ class AdjudicationsMigrationService(
 
         try {
           val mapping = adjudicationsService.createAdjudication(nomisAdjudication.toAdjudication())
-          createAdjudicationMapping(mapping, context = context)
+            ?: mappingCreator.createMigrationMappingResponse(nomisAdjudication)
 
-          telemetryClient.trackEvent(
-            "adjudications-migration-entity-migrated",
-            mapOf(
-              "adjudicationNumber" to adjudicationNumber.toString(),
-              "chargeSequence" to chargeSequence.toString(),
-              "prisonId" to nomisAdjudication.incident.prison.code,
-              "chargeNumber" to mapping.chargeNumberMapping.chargeNumber,
-              "offenderNo" to offenderNo,
-              "migrationId" to context.migrationId,
-            ),
-            null,
-          )
+          mapping?.run {
+            createAdjudicationMapping(mapping, context = context)
+
+            telemetryClient.trackEvent(
+              "adjudications-migration-entity-migrated",
+              mapOf(
+                "adjudicationNumber" to adjudicationNumber.toString(),
+                "chargeSequence" to chargeSequence.toString(),
+                "prisonId" to nomisAdjudication.incident.prison.code,
+                "chargeNumber" to mapping.chargeNumberMapping.chargeNumber,
+                "offenderNo" to offenderNo,
+                "migrationId" to context.migrationId,
+              ),
+              null,
+            )
+          } ?: run {
+            // unlikely to get here since we checked the mapping just above
+            telemetryClient.trackEvent(
+              "adjudications-migration-entity-already-migrated",
```

Comment:
I don't get it - looks like this gets run if the createAdjudication call returns a 406 - how does 406 = already migrated?

Reply:
DPS defines it so - they return 406 if that get a JPA duplicate key error

Reply:
Not mod security then 🤯